src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![no_std]
```

Comment:
I'm surprised we compile with default features enabled now - https://github.com/linebender/kurbo/blob/6ea9f53498749703606f6f8d019fe0bc9080ea30/src/lib.rs#L80 enables this conditionally

Clearly if CI passes, it does work though? Any explanation

Reply:
So, I removed the `features = "std"` leaving just `#![cfg_attr(not(test)), no_std)]` the thing that is going on here
is the `svg` module in lib.rs `#[cfg(feature = "std")] mod svg` and uses `std` freely rather than `core`/`alloc`.

This is part of the reason actually that `cargo test --no-default-features --features libm` breaks on kurbo :D
Because that should be `#[cfg(any(feature = "std", test)))] mod svg;` I think.

Reply:
Anyhow to summarize feeling here is that the `feature = "std"` part of that cfg_attr is specific to kurbo only conditionally exporting the svg module, and this combination becomes pretty fragile.  As evidenced by the fact that `cargo build --no-default-features --features libm` works but `cargo test ...` compile fails on a few things. So I would rather avoid that aspect.

The `test` part of the `cfg_attr` we may want to add now or in the future, but it doesn't currently affect us absent tests.

Reply:
So, this suggests that to test properly, we need to build Peniko with the `libm` feature on a platform without the standard library available?
Because at the moment (ignoring Kurbo) the reason that `features=std` works is because the std float functions are being used, but nothing about the feature itself actually enables those? Is there a cargo command to force non-usage of std on all targets?

Reply:
Ahh, now I see the problem you are indicating, that even though the lib defined `#![no_std]` and never imports std, the *kurbo* feature enabling causes `std` to be included thus `f32::sin` and friends work without error.

I don't see anything that looks solid beyond what you suggest (building for a target that doesn't have std).
There is some discussion and a PR here.
https://github.com/rust-lang/rust-clippy/issues/1570

Reply:
If it helps, that's why `kurbo` has a CI thing that builds for `thumbv7m-none-eabi`.

Reply:
It does, I had to go with `armv7a-none-eabi` because of `AtomicU64`, but tested that that fails with default features,
and succeeds with `--no-default-features --libm`.